controllers/controllers/services/instances/managed/controller_test.go

```diff
@@ -296,27 +283,181 @@ var _ = Describe("CFServiceInstance", func() {
-			brokerClient.ProvisionReturns(osbapi.ProvisionResponse{}, nil)
-		})
+			brokerClient.GetServiceInstanceLastOperationReturns(osbapi.LastOperationResponse{
+				State: "in-progress-or-whatever",
+			}, nil)
 
-		It("does not check last operation", func() {
-			Consistently(func(g Gomega) {
-				g.Expect(brokerClient.GetServiceInstanceLastOperationCallCount()).To(Equal(0))
-			}).Should(Succeed())
+			brokerClient.ProvisionReturns(osbapi.ProvisionResponse{
+				IsAsync:   true,
+				Operation: "operation-1",
+			}, nil)
 		})
 
-		It("set sets ready condition to true", func() {
+		It("set sets ready condition to false", func() {
 			Eventually(func(g Gomega) {
 				g.Expect(adminClient.Get(ctx, client.ObjectKeyFromObject(instance), instance)).To(Succeed())
 
 				g.Expect(instance.Status.Conditions).To(ContainElement(SatisfyAll(
 					HasType(Equal(korifiv1alpha1.StatusConditionReady)),
-					HasStatus(Equal(metav1.ConditionTrue)),
+					HasStatus(Equal(metav1.ConditionFalse)),
+					HasReason(Equal("ProvisionInProgress")),
 				)))
 			}).Should(Succeed())
 		})
+
+		It("sets in progress state in instance last operation", func() {
+			Eventually(func(g Gomega) {
+				g.Expect(adminClient.Get(ctx, client.ObjectKeyFromObject(instance), instance)).To(Succeed())
+				g.Expect(brokerClient.ProvisionCallCount()).To(BeNumerically(">=", 1))
+				g.Expect(instance.Status.LastOperation).To(Equal(services.LastOperation{
+					Type:  "create",
+					State: "in progress",
+				}))
+			}).Should(Succeed())
+		})
+
+		It("continuously checks the last operation", func() {
+			Eventually(func(g Gomega) {
+				g.Expect(brokerClient.GetServiceInstanceLastOperationCallCount()).To(BeNumerically(">", 1))
+				_, lastOp := brokerClient.GetServiceInstanceLastOperationArgsForCall(brokerClient.GetServiceInstanceLastOperationCallCount() - 1)
+				g.Expect(lastOp).To(Equal(osbapi.GetInstanceLastOperationRequest{
+					InstanceID: instance.Name,
+					GetLastOperationRequestParameters: osbapi.GetLastOperationRequestParameters{
+						ServiceId: "service-offering-id",
+						PlanID:    "service-plan-id",
+						Operation: "operation-1",
+					},
+				}))
+			}).Should(Succeed())
+		})
+
+		When("getting service last operation fails", func() {
+			BeforeEach(func() {
+				brokerClient.GetServiceInstanceLastOperationReturns(osbapi.LastOperationResponse{}, errors.New("get-last-op-failed"))
+			})
+
+			It("sets the ready condition to false", func() {
+				Eventually(func(g Gomega) {
+					g.Expect(adminClient.Get(ctx, client.ObjectKeyFromObject(instance), instance)).To(Succeed())
+
+					g.Expect(instance.Status.Conditions).To(ContainElement(SatisfyAll(
+						HasType(Equal(korifiv1alpha1.StatusConditionReady)),
+						HasStatus(Equal(metav1.ConditionFalse)),
+					)))
+				}).Should(Succeed())
+			})
+		})
+
+		When("the last operation is in progress", func() {
+			BeforeEach(func() {
+				brokerClient.GetServiceInstanceLastOperationReturns(osbapi.LastOperationResponse{
```

Comment:
This looks like it duplicates the test starting on line 288

Reply:
Indeed, the duplicate has been removed. Thanks!